binary_search_tree/binary_search_tree.py

```diff
@@ -1,3 +1,5 @@
+from collections import deque
```

Comment:
While it's important to learn to write these data structures, don't forget that Python makes some of them, and many others available to you- sometimes it's better to use other peoples solutions, as it saves you the time of generating your own.